hashintel/hash
# Persist the current workspace in local storage

## 🌟 What is the purpose of this PR?




This PR persists the currently active workspace (the workspace that's currently selected in the workspace switcher dropdown) in local storage, and decouples it's value from what the current account is in the URL.

This means entities/pages/entity types can be viewed in a workspace, wether or not they are owned by the corresponding user/org.

There are several implications to the changes introduced as part of this PR:
- when creating instances of an entity type, they are owned by the active account - not the account that owns the entity type
- when creating a new property type on an entity type, it is owned by the active account - not the account that owns the entity type
- creating a new page/entity type in the navbar creates a new page/entity type in the currently active workspace

## 🔗 Related links





- [Asana task](https://app.asana.com/0/1203312852763953/1203443872976450/f) _(internal)_

## 🔍 What does this change?




See commits.

## 📜 Does this require a change to the docs?



Not that I'm aware of.

## ⚠️ Known issues



- I don't understand the distinction between `useRouteNamespace` and `useRouteAccountInfo`, and what the longterm plan for these hooks is. My instinct is these should be merged into a single hook, as they both serve to parse the `/[account-slug]/...` from the current URL

## 🐾 Next steps



## 🛡 What tests cover this?



Unfortunately, none. Playwright is currently enabled, and is not fully covering the pages that are touched in this PR.

I've done my best to test things locally, but it's possible I may have missed something.

## ❓ How to test this?



1. Log in to the frontend
2. Try switching workspaces
3. Create pages/entities/entity types/property types and ensure the are being created in the currently active workspace

## 📹 Demo



https://user-images.githubusercontent.com/42802102/204879807-7412f4cb-2866-4429-8e99-1895604dbf0a.mov
